crates/router/tests/connectors/trustpay.rs

```diff
@@ -0,0 +1,304 @@
+            get_default_payment_info(),
+        )
+        .await
+        .unwrap();
+    assert_eq!(authorize_response.status, enums::AttemptStatus::Charged);
+}
+
+// Synchronizes a payment using the automatic capture flow (Non 3DS).
+#[actix_web::test]
+async fn should_sync_auto_captured_payment() {
+    let authorize_response = CONNECTOR
+        .make_payment(
+            get_default_payment_authorize_data(),
+            get_default_payment_info(),
+        )
+        .await
+        .unwrap();
+    assert_eq!(authorize_response.status, enums::AttemptStatus::Charged);
+    let txn_id = utils::get_connector_transaction_id(authorize_response.response);
+    assert_ne!(txn_id, None, "Empty connector transaction id");
+    let response = CONNECTOR
+        .psync_retry_till_status_matches(
+            enums::AttemptStatus::Charged,
+            Some(types::PaymentsSyncData {
+                connector_transaction_id: router::types::ResponseId::ConnectorTransactionId(
+                    txn_id.unwrap(),
+                ),
+                encoded_data: None,
+                capture_method: None,
+            }),
+            None,
+        )
+        .await
+        .unwrap();
+    assert_eq!(response.status, enums::AttemptStatus::Charged,);
+}
+
+// Refunds a payment using the automatic capture flow (Non 3DS).
+#[actix_web::test]
+async fn should_refund_auto_captured_payment() {
+    let response = CONNECTOR
+        .make_payment_and_refund(
+            get_default_payment_authorize_data(),
+            None,
+            get_default_payment_info(),
+        )
+        .await
+        .unwrap();
+    assert_eq!(
+        response.response.unwrap().refund_status,
+        enums::RefundStatus::Success,
+    );
+}
+
+// Partially refunds a payment using the automatic capture flow (Non 3DS).
+// #[actix_web::test]
+// async fn should_partially_refund_succeeded_payment() {
+//     let refund_response = CONNECTOR
+//         .make_payment_and_refund(
+//             get_default_payment_authorize_data(),
+//             Some(types::RefundsData {
+//                 refund_amount: 50,
+//                 ..utils::PaymentRefundType::default().0
+//             }),
+//             get_default_payment_info(),
+//         )
+//         .await
+//         .unwrap();
+//     assert_eq!(
+//         refund_response.response.unwrap().refund_status,
+//         enums::RefundStatus::Success,
+//     );
+// }
+
+// Creates multiple refunds against a payment using the automatic capture flow (Non 3DS).
+// #[actix_web::test]
+// async fn should_refund_succeeded_payment_multiple_times() {
+//     CONNECTOR
+//         .make_payment_and_multiple_refund(
+//             get_default_payment_authorize_data(),
+//             Some(types::RefundsData {
+//                 refund_amount: 50,
+//                 ..utils::PaymentRefundType::default().0
+//             }),
+//             get_default_payment_info(),
+//         )
+//         .await;
+// }
```

Comment:
remove these commented lines if not needed

Reply:
Done